lib/galaxy_test/api/test_metrics.py

```diff
@@ -0,0 +1,44 @@
+import json
+from datetime import datetime
+from unittest import mock
+
+from ._framework import ApiTestCase
+
+
+class MetricsApiTestCase(ApiTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._test_driver.app.trace_logger = mock.MagicMock()
+        self.mock_trace_logger = self._test_driver.app.trace_logger
```

Comment:
If you're going to mock out internals of Galaxy, I think this should be an integration test - https://github.com/galaxyproject/galaxy/tree/dev/test/integration. It needs to be tweaked a bit to make that happen.

Reply:
Thank you @jmchilton, that was my first approach. But then I thought that setting up the full [FluentTraceLogger](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/custom_logging/fluent_log.py) will be outside of the scope of the API so I just wanted to make sure the `trace_logger` was actually called the number of times that it was supposed to be. Now that I think of it, probably this was even too much for the API test and I should have simply asserted status code 200?
But you are right, I will try to move it to an integration test, and hopefully, that will cover also a bit of the `FluentTraceLogger` code in the test :)

Reply:
Nice hint @mvdbeek 😆 
So you mean creating the integration test with the real `FluentTraceLogger` and then try to test the API endpoint asynchronously?

Reply:
@jmchilton 
I have added the integration test along with the `fluent-logger` dev dependency and it seems to work, at least, the `FluentTraceLogger` gets called. Of course, there is no `fluentd` daemon running so nothing gets actually logged 😆 

@mvdbeek 
I am a bit confused with your async test proposal above, can you elaborate a bit on that? :)

Reply:
Sorry, that wasn't clear. The idea is that you test the API in isolation by setting (or ideally importing) the router you want to test. I'd say that is closer to what you'd do in a unit test. This is the pattern from https://fastapi.tiangolo.com/tutorial/testing/ / https://fastapi.tiangolo.com/advanced/async-tests/

Reply:
> The idea is that you test the API in isolation by setting (or ideally importing) the router you want to test. I'd say that is closer to what you'd do in a unit test.

I have made an attempt for this kind of test in https://github.com/galaxyproject/galaxy/pull/11568/commits/1315758c0945a40f6e0d1f18402efec2a633934f but I am not sure how to properly override or deal with the DI :/ I am getting:

````python
request = <starlette.requests.Request object at 0x7f422f092748>

    def _do_resolve(request: Request):
>       return get_app().resolve(dep_type)
E       AttributeError: 'NoneType' object has no attribute 'resolve'

lib/galaxy/webapps/galaxy/api/__init__.py:78: AttributeError
````

Obviously, because I haven't set up the DI but I don't know how to do it properly for the test 😅

Reply:
Uff, sorry for sending you down that rabbit hole ... maybe it was not a good idea and figuring this out is not in the scope of this PR